src/main/scala/com/advancedtelematic/campaigner/db/Campaigns.scala

```diff
@@ -222,45 +228,48 @@ protected [db] class Campaigns(implicit db: Database, ec: ExecutionContext)
           .result
           .map(_.toSet)
     }
+
+  /**
+   * Given a campaign and sets of accepted, scheduled and rejected
+   * devices, calculates and updates campaign and devices statuses in the database.
+   */
+  def updateCampaignAndDevicesStatuses(
+      campaign: Campaign,
+      acceptedDeviceIds: Set[DeviceId],
+      scheduledDeviceIds: Set[DeviceId],
+      rejectedDeviceIds: Set[DeviceId]): Future[Unit] = {
+    def persistDeviceUpdates(deviceIds: Set[DeviceId], status: DeviceStatus): DBIO[Unit] =
+      deviceUpdateRepo.persistManyAction(deviceIds.toSeq.map(deviceId =>
+        DeviceUpdate(campaign.id, campaign.updateId, deviceId, status)
+      ))
+
+    val action = for {
+      _ <- persistDeviceUpdates(acceptedDeviceIds, DeviceStatus.accepted)
+      _ <- persistDeviceUpdates(scheduledDeviceIds, DeviceStatus.scheduled)
+      _ <- persistDeviceUpdates(rejectedDeviceIds, DeviceStatus.rejected)
+      _ <- campaignStatusTransition.updateToCalculatedStatus(campaign.id)
```

Comment:
It would be interesting to check, what is a limit of a number of devices we can update in a single transaction

Reply:
If we hit a perf bottlenect, we can always switch to SQL :)

Reply:
It will not be a perf bottleneck, the transaction will fail, I guess...